tests/unit/ipv6_test.cc

```diff
@@ -103,3 +103,22 @@ SEASTAR_TEST_CASE(tcp_packet_test) {
     });
 }
 
+SEASTAR_TEST_CASE(ipv6_equal_test) {
+    uint16_t port{8080};
+    std::string const str_addr1{"abcd:fedc:ba98:7654:3210:0123:4567:89ab"};
```

Comment:
could you please put `const` specifier before `std::string`. or just drop it. just to be more consistent with the seastar code base.

and could you take this opportunity to add a test for comparing two addresses with the same IPv6 address but with different ports?